{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"collapsed_sections":["WhA-9c2LfjHT","AuPZs-jJfqNy","eTxjdFWrfuD4","w7x9aqeWsZiR","DVobJDZZlPpZ","v7QexSSD-6IX"],"mount_file_id":"1zxptoo9S_iB-AHmTbUfoZPDCnl0QUHDb","authorship_tag":"ABX9TyMYFa2oTRdUN1LaZcIMvYLy"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["# settings"],"metadata":{"id":"WhA-9c2LfjHT"}},{"cell_type":"code","source":["import os"],"metadata":{"id":"6SuZtkxU2W79","executionInfo":{"status":"ok","timestamp":1704201786468,"user_tz":-480,"elapsed":530,"user":{"displayName":"何子安","userId":"06861626017998849506"}}},"execution_count":2,"outputs":[]},{"cell_type":"code","source":["# my working directory\n","wd = \"/content/drive/MyDrive/NCCU_courses/112-1/Bioinformatics/readings/TADs\""],"metadata":{"id":"6n8snmcU2E53","executionInfo":{"status":"ok","timestamp":1704201786468,"user_tz":-480,"elapsed":4,"user":{"displayName":"何子安","userId":"06861626017998849506"}}},"execution_count":3,"outputs":[]},{"cell_type":"code","source":["os.listdir(wd)"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"2CAgiKfk2Ywn","executionInfo":{"status":"ok","timestamp":1704201787579,"user_tz":-480,"elapsed":1115,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"e7b44799-0890-405e-bebd-c5082752e6ac"},"execution_count":4,"outputs":[{"output_type":"execute_result","data":{"text/plain":["['test_fanc.ipynb',\n"," 'TADs are 3D structural units of higher-order chromosome organization in Drosophila.pdf',\n"," 'sample_data',\n"," 'final_proposal.pdf',\n"," 'final_proposal.gslides',\n"," 'examples',\n"," '.vscode',\n"," 'examples2',\n"," 'own_data']"]},"metadata":{},"execution_count":4}]},{"cell_type":"code","source":["!pwd\n","!ls"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"dYCqmHb-7Xtb","executionInfo":{"status":"ok","timestamp":1704201787580,"user_tz":-480,"elapsed":14,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"cf57f263-463a-4ee1-b79a-b2cf5a469c43"},"execution_count":5,"outputs":[{"output_type":"stream","name":"stdout","text":["/content\n","drive  sample_data\n"]}]},{"cell_type":"code","source":["# ! cp -r examples/ examples2/\n","! ls"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"OzeYJfA37mIF","executionInfo":{"status":"ok","timestamp":1704201787582,"user_tz":-480,"elapsed":15,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"fd35a23a-0f57-4a51-e92b-5afb41876dd3"},"execution_count":6,"outputs":[{"output_type":"stream","name":"stdout","text":["drive  sample_data\n"]}]},{"cell_type":"code","source":["# make change of my working directory\n","os.chdir(wd)"],"metadata":{"id":"APyQq2rx2a7k","executionInfo":{"status":"ok","timestamp":1704201787583,"user_tz":-480,"elapsed":14,"user":{"displayName":"何子安","userId":"06861626017998849506"}}},"execution_count":7,"outputs":[]},{"cell_type":"markdown","source":["download examplde data if you haven't:\n","\n","https://fan-c.readthedocs.io/en/latest/getting_started.html#example-analysis"],"metadata":{"id":"urAxYIItFeTc"}},{"cell_type":"markdown","source":["# installation"],"metadata":{"id":"AuPZs-jJfqNy"}},{"cell_type":"code","source":["# !wget -O examples.zip \"https://keeper.mpdl.mpg.de/d/147906745b634c779ed3/files/?p=/examples.zip&dl=1\"\n","# !unzip examples.zip"],"metadata":{"id":"K39pbQq93DLw","executionInfo":{"status":"ok","timestamp":1704201787583,"user_tz":-480,"elapsed":13,"user":{"displayName":"何子安","userId":"06861626017998849506"}}},"execution_count":8,"outputs":[]},{"cell_type":"code","source":["# act as `cd`\n","os.chdir(wd + \"/examples\")\n","!pwd"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"SVz8c7w63klh","executionInfo":{"status":"ok","timestamp":1704201788000,"user_tz":-480,"elapsed":429,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"6e4f63b6-6ddd-49a5-ea36-a295f93a41d6"},"execution_count":9,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/NCCU_courses/112-1/Bioinformatics/readings/TADs/examples\n"]}]},{"cell_type":"code","source":["# !pip uninstall -y biopython\n","!pip install -q biopython==1.75\n","!pip install -q fanc==0.9.27\n","# !pip uninstall -y fanc\n","# !pip install --pre -q fanc==0.9.26"],"metadata":{"id":"Yutf3LN73EaU","executionInfo":{"status":"ok","timestamp":1704201876601,"user_tz":-480,"elapsed":88606,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"fbb78333-ed71-445c-fa82-055eab977b56"},"execution_count":10,"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m16.4/16.4 MB\u001b[0m \u001b[31m44.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for biopython (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m281.4/281.4 kB\u001b[0m \u001b[31m2.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m21.9/21.9 MB\u001b[0m \u001b[31m41.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m44.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m209.7/209.7 kB\u001b[0m \u001b[31m16.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m41.6/41.6 kB\u001b[0m \u001b[31m3.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m103.7/103.7 kB\u001b[0m \u001b[31m10.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.9/1.9 MB\u001b[0m \u001b[31m65.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m134.8/134.8 kB\u001b[0m \u001b[31m13.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m137.9/137.9 kB\u001b[0m \u001b[31m2.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m115.3/115.3 kB\u001b[0m \u001b[31m11.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Building wheel for fanc (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for genomic_regions (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for gridmap (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for intervaltree (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for asciitree (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Building wheel for drmaa (setup.py) ... \u001b[?25l\u001b[?25hdone\n"]}]},{"cell_type":"code","source":["!pip list | grep fanc"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"nPPC42m8eX_X","executionInfo":{"status":"ok","timestamp":1704201877948,"user_tz":-480,"elapsed":1358,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"e3d3c23b-1ee3-419b-c083-8d0a54a209cc"},"execution_count":11,"outputs":[{"output_type":"stream","name":"stdout","text":["fanc                             0.9.27\n"]}]},{"cell_type":"code","source":["# fanc requires bowtie2\n","!apt install -y bowtie2"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"lFrLYzvN58lH","executionInfo":{"status":"ok","timestamp":1704201886492,"user_tz":-480,"elapsed":8549,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"17f7ebea-14dd-4352-d0bf-7029c3064367"},"execution_count":12,"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","Suggested packages:\n","  bowtie2-examples\n","The following NEW packages will be installed:\n","  bowtie2\n","0 upgraded, 1 newly installed, 0 to remove and 24 not upgraded.\n","Need to get 1,169 kB of archives.\n","After this operation, 3,735 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bowtie2 amd64 2.4.4-1 [1,169 kB]\n","Fetched 1,169 kB in 1s (1,140 kB/s)\n","Selecting previously unselected package bowtie2.\n","(Reading database ... 121658 files and directories currently installed.)\n","Preparing to unpack .../bowtie2_2.4.4-1_amd64.deb ...\n","Unpacking bowtie2 (2.4.4-1) ...\n","Setting up bowtie2 (2.4.4-1) ...\n","Processing triggers for man-db (2.10.2-1) ...\n"]}]},{"cell_type":"markdown","source":["# testing"],"metadata":{"id":"eTxjdFWrfuD4"}},{"cell_type":"code","source":["!ls -al"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"5nOOuq6-OESO","executionInfo":{"status":"ok","timestamp":1703847636801,"user_tz":-480,"elapsed":334,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"59651648-d10f-4053-d3e0-c79f14e16a27"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["total 570451\n","drwx------ 2 root root      4096 Dec 24 12:54 architecture\n","drwx------ 2 root root      4096 Dec 24 12:53 bwa-index\n","drwx------ 2 root root      4096 Dec 24 12:55 hg19_chr18_19\n","-rw------- 1 root root 139950370 Feb  4  2020 hg19_chr18_19.fa\n","-rw------- 1 root root   1019628 Feb  4  2020 hg19_chr18_19_re_fragments.bed\n","drwx------ 2 root root      4096 Dec 24 12:55 hicpro\n","drwx------ 7 root root      4096 Dec 27 17:11 output\n","-rw------- 1 root root 208957697 Feb  4  2020 SRR4271982_chr18_19_1.fastq.gzip\n","-rw------- 1 root root 209079585 Feb  4  2020 SRR4271982_chr18_19_2.fastq.gzip\n","-rw------- 1 root root     53056 Feb  4  2020 test.cool\n","-rw------- 1 root root  25059062 Feb  4  2020 test.hic\n"]}]},{"cell_type":"markdown","source":["https://fan-c.readthedocs.io/en/latest/getting_started.html#example-analysis"],"metadata":{"id":"Plt-XyQB-dP3"}},{"cell_type":"code","source":["# testing, output only processes\n","!fanc auto SRR4271982_chr18_19_1.fastq.gzip SRR4271982_chr18_19_2.fastq.gzip \\\n","    output/ \\\n","    -g hg19_chr18_19.fa \\\n","    -i hg19_chr18_19/hg19_chr18_19 \\\n","    -n fanc_example \\\n","    -t 4 \\\n","    --fanc-parallel \\\n","    -r HindIII \\\n","    --split-ligation-junction \\\n","    -q 30 \\\n","    -b 1mb, 50kb \\\n","    -f \\\n","    --run-with test"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"xzKiTbb436KU","executionInfo":{"status":"ok","timestamp":1703927463481,"user_tz":-480,"elapsed":2439,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"06339c81-b5df-44e1-b075-6ef1ab47f2a4"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["2023-12-30 09:11:02,761 INFO FAN-C version: 0.9.27\n","2023-12-30 09:11:02,791 INFO Output folder: output/\n","2023-12-30 09:11:02,791 INFO Input files: SRR4271982_chr18_19_1.fastq.gzip, SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-30 09:11:02,791 INFO Input file types: fastq, fastq\n","2023-12-30 09:11:02,791 INFO Final basename: fanc_example (you can change this with the -n option!)\n","0 (threads: 4, depends on: ): fanc map -m 25 -s 3 -t 4 --fanc-parallel --no-iterative --restriction-enzyme HindIII SRR4271982_chr18_19_1.fastq.gzip hg19_chr18_19/hg19_chr18_19 output/sam/SRR4271982_chr18_19_1.bam\n","1 (threads: 4, depends on: ): fanc map -m 25 -s 3 -t 4 --fanc-parallel --no-iterative --restriction-enzyme HindIII SRR4271982_chr18_19_2.fastq.gzip hg19_chr18_19/hg19_chr18_19 output/sam/SRR4271982_chr18_19_2.bam\n","2 (threads: 1, depends on: 0, 1): fanc sort_sam -t 4 --no-sambamba output/sam/SRR4271982_chr18_19_1.bam\n","3 (threads: 1, depends on: 0, 1): fanc sort_sam -t 4 --no-sambamba output/sam/SRR4271982_chr18_19_2.bam\n","4 (threads: 4, depends on: 2, 3): fanc pairs -f -g hg19_chr18_19.fa -t 4 -us -r HindIII -q 30.0 -S output/sam/SRR4271982_chr18_19_1.bam output/sam/SRR4271982_chr18_19_2.bam output/pairs/fanc_example.pairs\n","5 (threads: 1, depends on: 4): fanc pairs -d 10000 -l -p 2 --statistics-plot output/plots/stats/fanc_example.pairs.stats.pdf output/pairs/fanc_example.pairs\n","6 (threads: 1, depends on: 5): fanc pairs --ligation-error-plot output/plots/stats/fanc_example.pairs.ligation_error.pdf output/pairs/fanc_example.pairs\n","7 (threads: 1, depends on: 5): fanc pairs --re-dist-plot output/plots/stats/fanc_example.pairs.re_dist.pdf output/pairs/fanc_example.pairs\n","8 (threads: 1, depends on: 5): fanc hic -f output/pairs/fanc_example.pairs output/hic/fanc_example.hic\n","9 (threads: 4, depends on: 8): fanc hic -f -b 1000000 -r 0.1 -t 4 --statistics-plot output/plots/stats/fanc_example_1mb.stats.pdf -n --norm-method kr output/hic/fanc_example.hic output/hic/binned/fanc_example_1mb.hic\n","10 (threads: 4, depends on: 8): fanc hic -f -b 50000 -r 0.1 -t 4 --statistics-plot output/plots/stats/fanc_example_50kb.stats.pdf -n --norm-method kr output/hic/fanc_example.hic output/hic/binned/fanc_example_50kb.hic\n"]}]},{"cell_type":"markdown","source":["Parameters:\n","\n","https://vaquerizaslab.github.io/fanc/fanc-executable/fanc-generate-hic/fanc_auto.html?highlight=bowtie2#parameters\n"],"metadata":{"id":"96HlYCZG-fW-"}},{"cell_type":"code","source":["# actually run the pipeline\n","# take around 35 minutes\n","!fanc auto SRR4271982_chr18_19_1.fastq.gzip SRR4271982_chr18_19_2.fastq.gzip \\\n","    output/ \\\n","    -g hg19_chr18_19.fa \\\n","    -i hg19_chr18_19/hg19_chr18_19 \\\n","    -n fanc_example \\\n","    -t 4 \\\n","    --fanc-parallel \\\n","    -r HindIII \\\n","    --split-ligation-junction \\\n","    -q 30 \\\n","    -b 1mb, 50kb \\\n","    -f"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"TjOD8oBV53JK","executionInfo":{"status":"ok","timestamp":1703848683780,"user_tz":-480,"elapsed":976114,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"11a4167b-20c7-4226-be54-acc14774521e"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["2023-12-29 11:01:48,653 INFO FAN-C version: 0.9.27\n","2023-12-29 11:01:48,661 INFO Output folder: output/\n","2023-12-29 11:01:48,661 INFO Input files: SRR4271982_chr18_19_1.fastq.gzip, SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-29 11:01:48,662 INFO Input file types: fastq, fastq\n","2023-12-29 11:01:48,662 INFO Final basename: fanc_example (you can change this with the -n option!)\n","2023-12-29 11:01:48,662 INFO Creating output folders...\n","2023-12-29 11:01:50,056 INFO FAN-C version: 0.9.27\n","2023-12-29 11:01:50,058 INFO Starting mapping for SRR4271982_chr18_19_1.fastq.gzip\n","2023-12-29 11:01:50,182 INFO Starting to output alignments to intermediate SAM file /tmp/tmpwrxob80v/intermediate.sam\n","2023-12-29 11:06:29,351 INFO Converting intermediate SAM file to BAM (output/sam/SRR4271982_chr18_19_1.bam)\n","2023-12-29 11:06:52,050 INFO Success, removing intermediate.\n","2023-12-29 11:06:54,644 INFO FAN-C version: 0.9.27\n","2023-12-29 11:06:54,647 INFO Starting mapping for SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-29 11:06:54,748 INFO Starting to output alignments to intermediate SAM file /tmp/tmpxpk606au/intermediate.sam\n","2023-12-29 11:11:24,486 INFO Converting intermediate SAM file to BAM (output/sam/SRR4271982_chr18_19_2.bam)\n","2023-12-29 11:11:47,105 INFO Success, removing intermediate.\n","2023-12-29 11:11:49,137 INFO FAN-C version: 0.9.27\n","2023-12-29 11:11:49,147 INFO FAN-C version: 0.9.27\n","2023-12-29 11:12:22,259 INFO Replacing input SAM/BAM file output/sam/SRR4271982_chr18_19_2.bam with sorted version...\n","2023-12-29 11:12:22,264 INFO Replacing input SAM/BAM file output/sam/SRR4271982_chr18_19_1.bam with sorted version...\n","2023-12-29 11:12:25,718 INFO FAN-C version: 0.9.27\n","2023-12-29 11:12:25,719 INFO Getting genome regions (fragments or bins)\n","2023-12-29 11:12:25,719 INFO Getting regions\n","2023-12-29 11:12:37,207 INFO Three arguments detected, assuming SAM/BAM input.\n","2023-12-29 11:12:37,215 INFO Using filters appropriate for Bowtie2.\n","Buffers 100% (630 of 630) |###################################| Elapsed Time: 0:00:03 Time:  0:00:03\n","2023-12-29 11:14:16,497 INFO Updating region mappability\n","Mappability 100% (1611978 of 1611978) |#######################| Elapsed Time: 0:00:07 Time:  0:00:07\n","2023-12-29 11:14:31,144 INFO All done.\n","Closing remaining open files:KOTYJ6...done\n","2023-12-29 11:14:33,330 INFO FAN-C version: 0.9.27\n","2023-12-29 11:14:33,331 INFO One argument received, assuming existing Pairs object.\n","2023-12-29 11:14:33,399 INFO Filtering reads with RE distance > 10000bp\n","2023-12-29 11:14:33,400 INFO Filtering self-ligated read pairs\n","2023-12-29 11:14:33,401 INFO Filtering PCR duplicates, threshold <= 2bp\n","2023-12-29 11:14:42,235 INFO PCR duplicate stats: 186092 (11.5%) of pairs marked as duplicate.  (multiplicity:occurances) 3:15319 2:149430 4:1522 5:206 10:6 12:2 6:43 7:20 8:6 9:4 14:2\n","2023-12-29 11:14:42,235 INFO Running filters...\n","Filter 100% (630 of 630) |####################################| Elapsed Time: 0:00:38 Time:  0:00:38\n","2023-12-29 11:15:27,547 INFO Total: 1611978. Valid: 665942\n","2023-12-29 11:15:27,548 INFO Updating region mappability\n","Mappability 100% (665942 of 665942) |#########################| Elapsed Time: 0:00:08 Time:  0:00:08\n","2023-12-29 11:15:42,283 INFO Done.\n","2023-12-29 11:15:55,473 INFO Saving statistics...\n","2023-12-29 11:15:56,989 INFO All done.\n","2023-12-29 11:16:00,424 INFO FAN-C version: 0.9.27\n","2023-12-29 11:16:00,573 INFO FAN-C version: 0.9.27\n","2023-12-29 11:16:00,577 INFO One argument received, assuming existing Pairs object.\n","2023-12-29 11:16:00,792 INFO FAN-C version: 0.9.27\n","2023-12-29 11:16:00,794 INFO One argument received, assuming existing Pairs object.\n","2023-12-29 11:16:02,284 INFO Converting Pairs files\n","2023-12-29 11:16:02,383 INFO Converting output/pairs/fanc_example.pairs to Hic\n","/usr/local/lib/python3.10/dist-packages/fanc/plotting/statistics.py:179: UserWarning: \n","\n","`distplot` is a deprecated function and will be removed in seaborn v0.14.0.\n","\n","Please adapt your code to use either `displot` (a figure-level function with\n","similar flexibility) or `histplot` (an axes-level function for histograms).\n","\n","For a guide to updating your code to use the new functions, please see\n","https://gist.github.com/mwaskom/de44147ed2974457ad6372750bbe5751\n","\n","  sns.distplot(distances, ax=ax, color=color, **kwargs)\n","2023-12-29 11:16:10,016 INFO All done.\n","Hi-C convert 100% (665942 of 665942) |########################| Elapsed Time: 0:00:38 Time:  0:00:38\n","Ligation error 100% (1611978 of 1611978) |####################| Elapsed Time: 0:00:40 Time:  0:00:40\n","2023-12-29 11:17:10,138 INFO Pairs: 665942\n","2023-12-29 11:17:10,138 INFO Inter-chromosomal: 15237\n","2023-12-29 11:17:10,138 INFO Same fragment: 841237\n","2023-12-29 11:17:10,138 INFO Same: 308365\n","2023-12-29 11:17:10,138 INFO Inward: 162735\n","2023-12-29 11:17:10,138 INFO Outward: 154043\n","2023-12-29 11:17:12,572 INFO All done.\n","2023-12-29 11:17:13,969 INFO FAN-C version: 0.9.27\n","2023-12-29 11:17:14,036 INFO Binning Hic file (1000000)\n","2023-12-29 11:17:14,036 INFO Constructing binned genome...\n","2023-12-29 11:17:14,207 INFO Binning edges...\n","2023-12-29 11:17:14,933 INFO Binning Hi-C contacts\n","2023-12-29 11:17:15,257 INFO Launching processes\n","2023-12-29 11:17:20,349 INFO Submitting partitions\n","2023-12-29 11:17:20,378 INFO Collecting results\n","Binning 100% (3 of 3) |#######################################| Elapsed Time: 0:00:02 Time:  0:00:02\n","Buffers 100% (1 of 1) |#######################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Expected 100% (8434 of 8434) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-29 11:17:29,263 INFO Filtering low-coverage bins using absolute cutoff 0.0, relative cutoff 10.0%\n","2023-12-29 11:17:29,316 INFO Final absolute cutoff threshold is 872.9\n","2023-12-29 11:17:29,316 INFO Selected a total of 6 (4.3%) regions to be masked\n","2023-12-29 11:17:29,316 INFO Running filters...\n","Filter 100% (1 of 1) |########################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-29 11:17:29,350 INFO Total: 8434. Filtered: 143\n","Expected 100% (8291 of 8291) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-29 11:17:29,424 INFO Done.\n","2023-12-29 11:17:29,685 INFO Saving statistics...\n","2023-12-29 11:17:30,461 INFO Normalising binned Hic file\n","Expected 100% (8291 of 8291) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Closing remaining open files:output/hic/fanc_example.hic...done\n","2023-12-29 11:17:32,606 INFO FAN-C version: 0.9.27\n","2023-12-29 11:17:32,675 INFO Binning Hic file (50000)\n","2023-12-29 11:17:32,675 INFO Constructing binned genome...\n","2023-12-29 11:17:33,005 INFO Binning edges...\n","2023-12-29 11:17:33,739 INFO Binning Hi-C contacts\n","2023-12-29 11:17:34,042 INFO Launching processes\n","2023-12-29 11:17:38,383 INFO Submitting partitions\n","2023-12-29 11:17:38,406 INFO Collecting results\n","Binning 100% (3 of 3) |#######################################| Elapsed Time: 0:00:01 Time:  0:00:01\n","Buffers 100% (6 of 6) |#######################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Expected 100% (339969 of 339969) |############################| Elapsed Time: 0:00:01 Time:  0:00:01\n","2023-12-29 11:17:51,647 INFO Filtering low-coverage bins using absolute cutoff 0.0, relative cutoff 10.0%\n","2023-12-29 11:17:53,396 INFO Final absolute cutoff threshold is 44.45\n","2023-12-29 11:17:53,396 INFO Selected a total of 170 (6.2%) regions to be masked\n","2023-12-29 11:17:53,396 INFO Running filters...\n","Filter 100% (6 of 6) |########################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-29 11:17:54,185 INFO Total: 339969. Filtered: 1014\n","Expected 100% (338955 of 338955) |############################| Elapsed Time: 0:00:01 Time:  0:00:01\n","2023-12-29 11:17:56,175 INFO Done.\n","2023-12-29 11:17:56,558 INFO Saving statistics...\n","2023-12-29 11:17:57,273 INFO Normalising binned Hic file\n","Expected 100% (338955 of 338955) |############################| Elapsed Time: 0:00:01 Time:  0:00:01\n","Closing remaining open files:output/hic/fanc_example.hic...done\n"]}]},{"cell_type":"markdown","source":["# plotting"],"metadata":{"id":"w7x9aqeWsZiR"}},{"cell_type":"code","source":["!pwd"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ICAoWTxGpNHH","executionInfo":{"status":"ok","timestamp":1703848690699,"user_tz":-480,"elapsed":278,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"2747191d-283b-485b-f478-3a603c7fc41c"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/NCCU_courses/112-1/Bioinformatics/readings/TADs/examples\n"]}]},{"cell_type":"code","source":["!ls -al | sort\n","!echo\n","!ls -al output/hic/binned"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ksx_fLl4VY3K","executionInfo":{"status":"ok","timestamp":1703849432053,"user_tz":-480,"elapsed":757,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"ec1d49e1-8535-403d-f833-b485b85ce376"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["drwx------ 2 root root      4096 Dec 24 12:53 bwa-index\n","drwx------ 2 root root      4096 Dec 24 12:54 architecture\n","drwx------ 2 root root      4096 Dec 24 12:55 hg19_chr18_19\n","drwx------ 2 root root      4096 Dec 24 12:55 hicpro\n","drwx------ 7 root root      4096 Dec 27 17:11 output\n","-rw------- 1 root root   1019628 Feb  4  2020 hg19_chr18_19_re_fragments.bed\n","-rw------- 1 root root 139950370 Feb  4  2020 hg19_chr18_19.fa\n","-rw------- 1 root root 208957697 Feb  4  2020 SRR4271982_chr18_19_1.fastq.gzip\n","-rw------- 1 root root 209079585 Feb  4  2020 SRR4271982_chr18_19_2.fastq.gzip\n","-rw------- 1 root root  25059062 Feb  4  2020 test.hic\n","-rw------- 1 root root     53056 Feb  4  2020 test.cool\n","-rw------- 1 root root     93840 Dec 29 11:24 theplot.png\n","total 570543\n","\n","total 46385\n","-rw------- 1 root root  2032512 Dec 27 17:38 fanc_example_100kb.hic\n","-rw------- 1 root root 10109017 Dec 27 17:42 fanc_example_10kb.hic\n","-rw------- 1 root root   571044 Dec 29 11:17 fanc_example_1mb.hic\n","-rw------- 1 root root  1033989 Dec 27 17:37 fanc_example_250kb.hic\n","-rw------- 1 root root  4509145 Dec 27 17:40 fanc_example_25kb.hic\n","-rw------- 1 root root   513806 Dec 27 17:36 fanc_example_2mb.hic\n","-rw------- 1 root root   706229 Dec 27 17:37 fanc_example_500kb.hic\n","-rw------- 1 root root  3016905 Dec 29 11:18 fanc_example_50kb.hic\n","-rw------- 1 root root 24513051 Dec 27 17:46 fanc_example_5kb.hic\n","-rw------- 1 root root   489947 Dec 27 17:36 fanc_example_5mb.hic\n"]}]},{"cell_type":"code","source":["!fancplot chr18:20mb-100mb -o theplot.png -p square -vmax 0.05 output/hic/binned/fanc_example_25kb.hic"],"metadata":{"id":"Viw33lPQ-ojS","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1703849084740,"user_tz":-480,"elapsed":5771,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"0ac565c6-8956-4519-b68c-ad72530ceda6"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["2023-12-29 11:24:40,063 INFO Using non-interactive backend\n","2023-12-29 11:24:40,651 INFO Found 1 regions\n","Closing remaining open files:output/hic/binned/fanc_example_25kb.hic...done\n"]}]},{"cell_type":"markdown","source":["# testing parameters"],"metadata":{"id":"DVobJDZZlPpZ"}},{"cell_type":"markdown","source":["different restriction-enzyme: HindIII, DpnII\n","\n","(recognize and cut DNA sequences at different specific sites)"],"metadata":{"id":"4Ghf_-qale0f"}},{"cell_type":"code","source":["# testing, output only processes\n","!fanc auto SRR4271982_chr18_19_1.fastq.gzip SRR4271982_chr18_19_2.fastq.gzip \\\n","    test_params_output/ \\\n","    -g hg19_chr18_19.fa \\\n","    -i hg19_chr18_19/hg19_chr18_19 \\\n","    -n fanc_example \\\n","    -t 7 \\\n","    --fanc-parallel \\\n","    -r DpnII \\\n","    --split-ligation-junction \\\n","    -q 30 \\\n","    -b 1mb, 50kb \\\n","    -f \\\n","    --run-with test"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"KmzHwe3TlKpI","executionInfo":{"status":"ok","timestamp":1703920769105,"user_tz":-480,"elapsed":1941,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"e7679a33-f688-424f-83cf-24949962e06d"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["2023-12-30 07:19:28,395 INFO FAN-C version: 0.9.27\n","2023-12-30 07:19:28,405 INFO Output folder: test_params_output/\n","2023-12-30 07:19:28,405 INFO Input files: SRR4271982_chr18_19_1.fastq.gzip, SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-30 07:19:28,405 INFO Input file types: fastq, fastq\n","2023-12-30 07:19:28,405 INFO Final basename: fanc_example (you can change this with the -n option!)\n","0 (threads: 7, depends on: ): fanc map -m 25 -s 3 -t 7 --fanc-parallel --no-iterative --restriction-enzyme DpnII SRR4271982_chr18_19_1.fastq.gzip hg19_chr18_19/hg19_chr18_19 test_params_output/sam/SRR4271982_chr18_19_1.bam\n","1 (threads: 7, depends on: ): fanc map -m 25 -s 3 -t 7 --fanc-parallel --no-iterative --restriction-enzyme DpnII SRR4271982_chr18_19_2.fastq.gzip hg19_chr18_19/hg19_chr18_19 test_params_output/sam/SRR4271982_chr18_19_2.bam\n","2 (threads: 1, depends on: 0, 1): fanc sort_sam -t 4 --no-sambamba test_params_output/sam/SRR4271982_chr18_19_1.bam\n","3 (threads: 1, depends on: 0, 1): fanc sort_sam -t 4 --no-sambamba test_params_output/sam/SRR4271982_chr18_19_2.bam\n","4 (threads: 7, depends on: 2, 3): fanc pairs -f -g hg19_chr18_19.fa -t 7 -us -r DpnII -q 30.0 -S test_params_output/sam/SRR4271982_chr18_19_1.bam test_params_output/sam/SRR4271982_chr18_19_2.bam test_params_output/pairs/fanc_example.pairs\n","5 (threads: 1, depends on: 4): fanc pairs -d 10000 -l -p 2 --statistics-plot test_params_output/plots/stats/fanc_example.pairs.stats.pdf test_params_output/pairs/fanc_example.pairs\n","6 (threads: 1, depends on: 5): fanc pairs --ligation-error-plot test_params_output/plots/stats/fanc_example.pairs.ligation_error.pdf test_params_output/pairs/fanc_example.pairs\n","7 (threads: 1, depends on: 5): fanc pairs --re-dist-plot test_params_output/plots/stats/fanc_example.pairs.re_dist.pdf test_params_output/pairs/fanc_example.pairs\n","8 (threads: 1, depends on: 5): fanc hic -f test_params_output/pairs/fanc_example.pairs test_params_output/hic/fanc_example.hic\n","9 (threads: 4, depends on: 8): fanc hic -f -b 1000000 -r 0.1 -t 4 --statistics-plot test_params_output/plots/stats/fanc_example_1mb.stats.pdf -n --norm-method kr test_params_output/hic/fanc_example.hic test_params_output/hic/binned/fanc_example_1mb.hic\n","10 (threads: 4, depends on: 8): fanc hic -f -b 50000 -r 0.1 -t 4 --statistics-plot test_params_output/plots/stats/fanc_example_50kb.stats.pdf -n --norm-method kr test_params_output/hic/fanc_example.hic test_params_output/hic/binned/fanc_example_50kb.hic\n"]}]},{"cell_type":"code","source":["!pwd\n","!ls -a | grep test_params\n","# !rm -r test_params_output/"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"yGKUJyC54k32","executionInfo":{"status":"ok","timestamp":1703925726058,"user_tz":-480,"elapsed":756,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"270650a7-1c76-4310-9bfe-7029d6aaad3d"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/NCCU_courses/112-1/Bioinformatics/readings/TADs/examples\n","test_params_output\n"]}]},{"cell_type":"code","source":["# testing, output only processes\n","!fanc auto SRR4271982_chr18_19_1.fastq.gzip SRR4271982_chr18_19_2.fastq.gzip \\\n","    test_params_output/ \\\n","    -g hg19_chr18_19.fa \\\n","    -i hg19_chr18_19/hg19_chr18_19 \\\n","    -n fanc_example \\\n","    -t 7 \\\n","    --fanc-parallel \\\n","    -r DpnII \\\n","    --split-ligation-junction \\\n","    -q 30 \\\n","    -b 1mb, 50kb \\\n","    -f"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"8X47G2-Dl4yf","executionInfo":{"status":"ok","timestamp":1703924150675,"user_tz":-480,"elapsed":3330287,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"56dd7535-8f35-41b4-b6b7-f0d08934f695"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["2023-12-30 07:20:21,623 INFO FAN-C version: 0.9.27\n","2023-12-30 07:20:21,632 INFO Output folder: test_params_output/\n","2023-12-30 07:20:21,633 INFO Input files: SRR4271982_chr18_19_1.fastq.gzip, SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-30 07:20:21,633 INFO Input file types: fastq, fastq\n","2023-12-30 07:20:21,633 INFO Final basename: fanc_example (you can change this with the -n option!)\n","2023-12-30 07:20:21,633 INFO Creating output folders...\n","2023-12-30 07:20:23,153 INFO FAN-C version: 0.9.27\n","2023-12-30 07:20:23,155 INFO Starting mapping for SRR4271982_chr18_19_1.fastq.gzip\n","2023-12-30 07:20:23,296 INFO Starting to output alignments to intermediate SAM file /tmp/tmpgyqk9rri/intermediate.sam\n","2023-12-30 07:26:47,684 INFO Converting intermediate SAM file to BAM (test_params_output/sam/SRR4271982_chr18_19_1.bam)\n","2023-12-30 07:27:13,721 INFO Success, removing intermediate.\n","2023-12-30 07:27:17,315 INFO FAN-C version: 0.9.27\n","2023-12-30 07:27:17,319 INFO Starting mapping for SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-30 07:27:17,483 INFO Starting to output alignments to intermediate SAM file /tmp/tmpods1ux32/intermediate.sam\n","2023-12-30 07:34:03,638 INFO Converting intermediate SAM file to BAM (test_params_output/sam/SRR4271982_chr18_19_2.bam)\n","2023-12-30 07:34:28,063 INFO Success, removing intermediate.\n","2023-12-30 07:34:32,311 INFO FAN-C version: 0.9.27\n","2023-12-30 07:34:32,608 INFO FAN-C version: 0.9.27\n","2023-12-30 07:35:16,740 INFO Replacing input SAM/BAM file test_params_output/sam/SRR4271982_chr18_19_2.bam with sorted version...\n","2023-12-30 07:35:18,138 INFO Replacing input SAM/BAM file test_params_output/sam/SRR4271982_chr18_19_1.bam with sorted version...\n","2023-12-30 07:35:20,647 INFO FAN-C version: 0.9.27\n","2023-12-30 07:35:20,649 INFO Getting genome regions (fragments or bins)\n","2023-12-30 07:35:20,649 INFO Getting regions\n","2023-12-30 07:35:40,839 INFO Three arguments detected, assuming SAM/BAM input.\n","2023-12-30 07:35:40,888 INFO Using filters appropriate for Bowtie2.\n","Buffers 100% (4936 of 4936) |#################################| Elapsed Time: 0:00:34 Time:  0:00:34\n","2023-12-30 07:43:14,643 INFO Updating region mappability\n","Mappability 100% (1604425 of 1604425) |#######################| Elapsed Time: 0:01:16 Time:  0:01:16\n","2023-12-30 07:45:50,530 INFO All done.\n","Closing remaining open files:NJKDXB...done\n","2023-12-30 07:45:53,123 INFO FAN-C version: 0.9.27\n","2023-12-30 07:45:53,125 INFO One argument received, assuming existing Pairs object.\n","2023-12-30 07:45:53,267 INFO Filtering reads with RE distance > 10000bp\n","2023-12-30 07:45:53,273 INFO Filtering self-ligated read pairs\n","2023-12-30 07:45:53,279 INFO Filtering PCR duplicates, threshold <= 2bp\n","2023-12-30 07:47:22,082 INFO PCR duplicate stats: 198168 (12.4%) of pairs marked as duplicate.  (multiplicity:occurances) 3:17138 2:156361 4:1781 5:233 10:4 6:43 9:6 7:20 14:2 8:5\n","2023-12-30 07:47:22,083 INFO Running filters...\n","Filter 100% (4936 of 4936) |##################################| Elapsed Time: 0:07:07 Time:  0:07:07\n","2023-12-30 07:56:01,032 INFO Total: 1604425. Valid: 963153\n","2023-12-30 07:56:01,033 INFO Updating region mappability\n","Mappability 100% (963153 of 963153) |#########################| Elapsed Time: 0:01:39 Time:  0:01:39\n","2023-12-30 07:59:28,782 INFO Done.\n","2023-12-30 08:01:57,032 INFO Saving statistics...\n","2023-12-30 08:01:58,478 INFO All done.\n","2023-12-30 08:02:04,279 INFO FAN-C version: 0.9.27\n","2023-12-30 08:02:04,287 INFO One argument received, assuming existing Pairs object.\n","2023-12-30 08:02:04,677 INFO FAN-C version: 0.9.27\n","2023-12-30 08:02:05,384 INFO FAN-C version: 0.9.27\n","2023-12-30 08:02:05,403 INFO One argument received, assuming existing Pairs object.\n","2023-12-30 08:02:07,700 INFO Converting Pairs files\n","2023-12-30 08:02:07,800 INFO Converting test_params_output/pairs/fanc_example.pairs to Hic\n","/usr/local/lib/python3.10/dist-packages/fanc/plotting/statistics.py:179: UserWarning: \n","\n","`distplot` is a deprecated function and will be removed in seaborn v0.14.0.\n","\n","Please adapt your code to use either `displot` (a figure-level function with\n","similar flexibility) or `histplot` (an axes-level function for histograms).\n","\n","For a guide to updating your code to use the new functions, please see\n","https://gist.github.com/mwaskom/de44147ed2974457ad6372750bbe5751\n","\n","  sns.distplot(distances, ax=ax, color=color, **kwargs)\n","2023-12-30 08:02:49,025 INFO All done.\n","Ligation error 100% (1604425 of 1604425) |####################| Elapsed Time: 0:03:17 Time:  0:03:17\n","2023-12-30 08:10:07,510 INFO Pairs: 963153\n","2023-12-30 08:10:07,510 INFO Inter-chromosomal: 15180\n","2023-12-30 08:10:07,510 INFO Same fragment: 493784\n","2023-12-30 08:10:07,510 INFO Same: 311492\n","2023-12-30 08:10:07,510 INFO Inward: 298666\n","2023-12-30 08:10:07,510 INFO Outward: 176893\n","Hi-C convert  79% (760891 of 963153) |##################      | Elapsed Time: 0:05:35 ETA:   0:03:332023-12-30 08:10:13,761 INFO All done.\n","Hi-C convert 100% (963153 of 963153) |########################| Elapsed Time: 0:06:24 Time:  0:06:24\n","2023-12-30 08:12:02,545 INFO FAN-C version: 0.9.27\n","2023-12-30 08:12:02,641 INFO Binning Hic file (1000000)\n","2023-12-30 08:12:02,642 INFO Constructing binned genome...\n","2023-12-30 08:12:05,046 INFO Binning edges...\n","2023-12-30 08:12:05,120 INFO Binning Hi-C contacts\n","2023-12-30 08:12:11,293 INFO Launching processes\n","2023-12-30 08:12:21,421 INFO Submitting partitions\n","2023-12-30 08:12:21,455 INFO Collecting results\n","Binning 100% (3 of 3) |#######################################| Elapsed Time: 0:00:50 Time:  0:00:50\n","Buffers 100% (1 of 1) |#######################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Expected 100% (8431 of 8431) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-30 08:13:42,506 INFO Filtering low-coverage bins using absolute cutoff 0.0, relative cutoff 10.0%\n","2023-12-30 08:13:42,584 INFO Final absolute cutoff threshold is 1.118e+03\n","2023-12-30 08:13:42,584 INFO Selected a total of 6 (4.3%) regions to be masked\n","2023-12-30 08:13:42,584 INFO Running filters...\n","Filter 100% (1 of 1) |########################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-30 08:13:42,633 INFO Total: 8431. Filtered: 141\n","Expected 100% (8290 of 8290) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-30 08:13:42,747 INFO Done.\n","2023-12-30 08:13:43,140 INFO Saving statistics...\n","2023-12-30 08:13:43,549 INFO Normalising binned Hic file\n","Expected 100% (8290 of 8290) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Closing remaining open files:test_params_output/hic/fanc_example.hic...done\n","2023-12-30 08:13:45,931 INFO FAN-C version: 0.9.27\n","2023-12-30 08:13:46,042 INFO Binning Hic file (50000)\n","2023-12-30 08:13:46,043 INFO Constructing binned genome...\n","2023-12-30 08:13:50,412 INFO Binning edges...\n","2023-12-30 08:13:50,575 INFO Binning Hi-C contacts\n","2023-12-30 08:13:55,306 INFO Launching processes\n","2023-12-30 08:14:05,529 INFO Submitting partitions\n","2023-12-30 08:14:05,575 INFO Collecting results\n","Binning 100% (3 of 3) |#######################################| Elapsed Time: 0:00:51 Time:  0:00:51\n","Buffers 100% (6 of 6) |#######################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Expected 100% (338766 of 338766) |############################| Elapsed Time: 0:00:03 Time:  0:00:03\n","2023-12-30 08:15:31,946 INFO Filtering low-coverage bins using absolute cutoff 0.0, relative cutoff 10.0%\n","2023-12-30 08:15:34,536 INFO Final absolute cutoff threshold is 56.75\n","2023-12-30 08:15:34,537 INFO Selected a total of 143 (5.2%) regions to be masked\n","2023-12-30 08:15:34,537 INFO Running filters...\n","Filter 100% (6 of 6) |########################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-30 08:15:35,735 INFO Total: 338766. Filtered: 312\n","Expected 100% (338454 of 338454) |############################| Elapsed Time: 0:00:02 Time:  0:00:02\n","2023-12-30 08:15:38,618 INFO Done.\n","2023-12-30 08:15:39,079 INFO Saving statistics...\n","2023-12-30 08:15:39,474 INFO Normalising binned Hic file\n","Expected 100% (338454 of 338454) |############################| Elapsed Time: 0:00:03 Time:  0:00:03\n","Closing remaining open files:test_params_output/hic/fanc_example.hic...done\n"]}]},{"cell_type":"markdown","source":["# testing necessary files"],"metadata":{"id":"v7QexSSD-6IX"}},{"cell_type":"code","source":["!pwd"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"_bvPqTLR_ANY","executionInfo":{"status":"ok","timestamp":1703929377176,"user_tz":-480,"elapsed":7,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"c5d25d02-33ee-4ac0-9fc3-ee426e72e7d9"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/NCCU_courses/112-1/Bioinformatics/readings/TADs/examples\n"]}]},{"cell_type":"code","source":["os.chdir(wd + \"/examples2\")"],"metadata":{"id":"sZqOlChv_jZe"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!ls -al"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ehY5fh9U_xWS","executionInfo":{"status":"ok","timestamp":1703929543394,"user_tz":-480,"elapsed":303,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"2e9008e9-c57f-4821-c743-4aec5e067f33"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["total 544915\n","drwx------ 2 root root      4096 Dec 30 09:00 hg19_chr18_19\n","-rw------- 1 root root 139950370 Dec 30 09:00 hg19_chr18_19.fa\n","-rw------- 1 root root 208957697 Dec 30 09:00 SRR4271982_chr18_19_1.fastq.gzip\n","-rw------- 1 root root 209079585 Dec 30 09:00 SRR4271982_chr18_19_2.fastq.gzip\n"]}]},{"cell_type":"code","source":["# testing, output only processes\n","!fanc auto SRR4271982_chr18_19_1.fastq.gzip SRR4271982_chr18_19_2.fastq.gzip \\\n","    test_necessary_files_output/ \\\n","    -g hg19_chr18_19.fa \\\n","    -i hg19_chr18_19/hg19_chr18_19 \\\n","    -n fanc_example \\\n","    -t 7 \\\n","    --fanc-parallel \\\n","    -r HindIII \\\n","    --split-ligation-junction \\\n","    -q 30 \\\n","    --iterative-quality-cutoff \\\n","    -b 1mb, 50kb \\\n","    -f \\\n","    --run-with test"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"vfCZYa-ymFyx","executionInfo":{"status":"ok","timestamp":1703929557936,"user_tz":-480,"elapsed":2217,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"b554ecc5-90aa-4815-9f0d-cdba1360f44e"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["2023-12-30 09:45:57,286 INFO FAN-C version: 0.9.27\n","2023-12-30 09:45:57,315 INFO Output folder: test_necessary_files_output/\n","2023-12-30 09:45:57,315 INFO Input files: SRR4271982_chr18_19_1.fastq.gzip, SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-30 09:45:57,315 INFO Input file types: fastq, fastq\n","2023-12-30 09:45:57,315 INFO Final basename: fanc_example (you can change this with the -n option!)\n","0 (threads: 7, depends on: ): fanc map -m 25 -s 3 -t 7 --fanc-parallel --no-iterative --restriction-enzyme HindIII SRR4271982_chr18_19_1.fastq.gzip hg19_chr18_19/hg19_chr18_19 test_necessary_files_output/sam/SRR4271982_chr18_19_1.bam\n","1 (threads: 7, depends on: ): fanc map -m 25 -s 3 -t 7 --fanc-parallel --no-iterative --restriction-enzyme HindIII SRR4271982_chr18_19_2.fastq.gzip hg19_chr18_19/hg19_chr18_19 test_necessary_files_output/sam/SRR4271982_chr18_19_2.bam\n","2 (threads: 1, depends on: 0, 1): fanc sort_sam -t 4 --no-sambamba test_necessary_files_output/sam/SRR4271982_chr18_19_1.bam\n","3 (threads: 1, depends on: 0, 1): fanc sort_sam -t 4 --no-sambamba test_necessary_files_output/sam/SRR4271982_chr18_19_2.bam\n","4 (threads: 7, depends on: 2, 3): fanc pairs -f -g hg19_chr18_19.fa -t 7 -us -r HindIII -q 30.0 -S test_necessary_files_output/sam/SRR4271982_chr18_19_1.bam test_necessary_files_output/sam/SRR4271982_chr18_19_2.bam test_necessary_files_output/pairs/fanc_example.pairs\n","5 (threads: 1, depends on: 4): fanc pairs -d 10000 -l -p 2 --statistics-plot test_necessary_files_output/plots/stats/fanc_example.pairs.stats.pdf test_necessary_files_output/pairs/fanc_example.pairs\n","6 (threads: 1, depends on: 5): fanc pairs --ligation-error-plot test_necessary_files_output/plots/stats/fanc_example.pairs.ligation_error.pdf test_necessary_files_output/pairs/fanc_example.pairs\n","7 (threads: 1, depends on: 5): fanc pairs --re-dist-plot test_necessary_files_output/plots/stats/fanc_example.pairs.re_dist.pdf test_necessary_files_output/pairs/fanc_example.pairs\n","8 (threads: 1, depends on: 5): fanc hic -f test_necessary_files_output/pairs/fanc_example.pairs test_necessary_files_output/hic/fanc_example.hic\n","9 (threads: 4, depends on: 8): fanc hic -f -b 1000000 -r 0.1 -t 4 --statistics-plot test_necessary_files_output/plots/stats/fanc_example_1mb.stats.pdf -n --norm-method kr test_necessary_files_output/hic/fanc_example.hic test_necessary_files_output/hic/binned/fanc_example_1mb.hic\n","10 (threads: 4, depends on: 8): fanc hic -f -b 50000 -r 0.1 -t 4 --statistics-plot test_necessary_files_output/plots/stats/fanc_example_50kb.stats.pdf -n --norm-method kr test_necessary_files_output/hic/fanc_example.hic test_necessary_files_output/hic/binned/fanc_example_50kb.hic\n"]}]},{"cell_type":"code","source":["# testing, output only processes\n","!fanc auto SRR4271982_chr18_19_1.fastq.gzip SRR4271982_chr18_19_2.fastq.gzip \\\n","    test_necessaray_files_output/ \\\n","    -g hg19_chr18_19.fa \\\n","    -i hg19_chr18_19/hg19_chr18_19 \\\n","    -n fanc_example \\\n","    -t 7 \\\n","    --fanc-parallel \\\n","    -r HindIII \\\n","    --split-ligation-junction \\\n","    -q 30 \\\n","    --iterative-quality-cutoff \\\n","    -b 1mb, 50kb \\\n","    -f"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"JYBZheqz_9tY","executionInfo":{"status":"ok","timestamp":1703931135278,"user_tz":-480,"elapsed":1571461,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"8a775f1f-4c0e-4d24-fd92-aecc205d1ca6"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["2023-12-30 09:46:05,945 INFO FAN-C version: 0.9.27\n","2023-12-30 09:46:05,957 INFO Output folder: test_necessaray_files_output/\n","2023-12-30 09:46:05,957 INFO Input files: SRR4271982_chr18_19_1.fastq.gzip, SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-30 09:46:05,957 INFO Input file types: fastq, fastq\n","2023-12-30 09:46:05,957 INFO Final basename: fanc_example (you can change this with the -n option!)\n","2023-12-30 09:46:05,957 INFO Creating output folders...\n","2023-12-30 09:46:08,444 INFO FAN-C version: 0.9.27\n","2023-12-30 09:46:08,449 INFO Starting mapping for SRR4271982_chr18_19_1.fastq.gzip\n","2023-12-30 09:46:08,655 INFO Starting to output alignments to intermediate SAM file /tmp/tmp56mbtj0c/intermediate.sam\n","2023-12-30 09:54:04,618 INFO Converting intermediate SAM file to BAM (test_necessaray_files_output/sam/SRR4271982_chr18_19_1.bam)\n","2023-12-30 09:54:31,874 INFO Success, removing intermediate.\n","2023-12-30 09:54:35,049 INFO FAN-C version: 0.9.27\n","2023-12-30 09:54:35,053 INFO Starting mapping for SRR4271982_chr18_19_2.fastq.gzip\n","2023-12-30 09:54:35,243 INFO Starting to output alignments to intermediate SAM file /tmp/tmpztuosda7/intermediate.sam\n","2023-12-30 10:02:16,745 INFO Converting intermediate SAM file to BAM (test_necessaray_files_output/sam/SRR4271982_chr18_19_2.bam)\n","2023-12-30 10:02:44,549 INFO Success, removing intermediate.\n","2023-12-30 10:02:47,453 INFO FAN-C version: 0.9.27\n","2023-12-30 10:02:47,472 INFO FAN-C version: 0.9.27\n","2023-12-30 10:03:36,641 INFO Replacing input SAM/BAM file test_necessaray_files_output/sam/SRR4271982_chr18_19_2.bam with sorted version...\n","2023-12-30 10:03:37,113 INFO Replacing input SAM/BAM file test_necessaray_files_output/sam/SRR4271982_chr18_19_1.bam with sorted version...\n","2023-12-30 10:03:40,373 INFO FAN-C version: 0.9.27\n","2023-12-30 10:03:40,375 INFO Getting genome regions (fragments or bins)\n","2023-12-30 10:03:40,375 INFO Getting regions\n","2023-12-30 10:03:54,029 INFO Three arguments detected, assuming SAM/BAM input.\n","2023-12-30 10:03:54,032 INFO Using filters appropriate for Bowtie2.\n","Buffers 100% (630 of 630) |###################################| Elapsed Time: 0:00:05 Time:  0:00:05\n","2023-12-30 10:06:27,502 INFO Updating region mappability\n","Mappability 100% (1611978 of 1611978) |#######################| Elapsed Time: 0:00:13 Time:  0:00:13\n","2023-12-30 10:06:50,872 INFO All done.\n","Closing remaining open files:2JRU89...done\n","2023-12-30 10:06:52,958 INFO FAN-C version: 0.9.27\n","2023-12-30 10:06:52,960 INFO One argument received, assuming existing Pairs object.\n","2023-12-30 10:06:53,025 INFO Filtering reads with RE distance > 10000bp\n","2023-12-30 10:06:53,027 INFO Filtering self-ligated read pairs\n","2023-12-30 10:06:53,028 INFO Filtering PCR duplicates, threshold <= 2bp\n","2023-12-30 10:07:07,151 INFO PCR duplicate stats: 186092 (11.5%) of pairs marked as duplicate.  (multiplicity:occurances) 3:15319 2:149430 4:1522 5:206 10:6 12:2 6:43 7:20 8:6 9:4 14:2\n","2023-12-30 10:07:07,151 INFO Running filters...\n","Filter 100% (630 of 630) |####################################| Elapsed Time: 0:00:54 Time:  0:00:54\n","2023-12-30 10:08:12,147 INFO Total: 1611978. Valid: 665942\n","2023-12-30 10:08:12,147 INFO Updating region mappability\n","Mappability 100% (665942 of 665942) |#########################| Elapsed Time: 0:00:10 Time:  0:00:10\n","2023-12-30 10:08:33,411 INFO Done.\n","2023-12-30 10:08:52,938 INFO Saving statistics...\n","2023-12-30 10:08:53,593 INFO All done.\n","2023-12-30 10:08:57,650 INFO FAN-C version: 0.9.27\n","2023-12-30 10:08:57,652 INFO One argument received, assuming existing Pairs object.\n","2023-12-30 10:08:57,706 INFO FAN-C version: 0.9.27\n","2023-12-30 10:08:57,797 INFO FAN-C version: 0.9.27\n","2023-12-30 10:08:57,799 INFO One argument received, assuming existing Pairs object.\n","2023-12-30 10:08:58,760 INFO Converting Pairs files\n","2023-12-30 10:08:58,839 INFO Converting test_necessaray_files_output/pairs/fanc_example.pairs to Hic\n","/usr/local/lib/python3.10/dist-packages/fanc/plotting/statistics.py:179: UserWarning: \n","\n","`distplot` is a deprecated function and will be removed in seaborn v0.14.0.\n","\n","Please adapt your code to use either `displot` (a figure-level function with\n","similar flexibility) or `histplot` (an axes-level function for histograms).\n","\n","For a guide to updating your code to use the new functions, please see\n","https://gist.github.com/mwaskom/de44147ed2974457ad6372750bbe5751\n","\n","  sns.distplot(distances, ax=ax, color=color, **kwargs)\n","2023-12-30 10:09:11,398 INFO All done.\n","Hi-C convert 100% (665942 of 665942) |########################| Elapsed Time: 0:01:01 Time:  0:01:01\n","Ligation error  81% (1305703 of 1611978) |################    | Elapsed Time: 0:01:09 ETA:   0:00:272023-12-30 10:10:47,101 INFO FAN-C version: 0.9.27\n","2023-12-30 10:10:47,614 INFO Binning Hic file (1000000)\n","2023-12-30 10:10:47,614 INFO Constructing binned genome...\n","Ligation error  83% (1337943 of 1611978) |################    | Elapsed Time: 0:01:12 ETA:   0:00:272023-12-30 10:10:49,820 INFO Binning edges...\n","2023-12-30 10:10:50,087 INFO Binning Hi-C contacts\n","Ligation error  84% (1354064 of 1611978) |################    | Elapsed Time: 0:01:16 ETA:   0:00:572023-12-30 10:10:53,557 INFO Launching processes\n","Ligation error 100% (1611978 of 1611978) |####################| Elapsed Time: 0:01:26 Time:  0:01:26\n","2023-12-30 10:11:02,741 INFO Pairs: 665942\n","2023-12-30 10:11:02,741 INFO Inter-chromosomal: 15237\n","2023-12-30 10:11:02,741 INFO Same fragment: 841237\n","2023-12-30 10:11:02,741 INFO Same: 308365\n","2023-12-30 10:11:02,741 INFO Inward: 162735\n","2023-12-30 10:11:02,741 INFO Outward: 154043\n","2023-12-30 10:11:06,448 INFO Submitting partitions\n","2023-12-30 10:11:06,509 INFO Collecting results\n","2023-12-30 10:11:08,049 INFO All done.\n","Binning 100% (3 of 3) |#######################################| Elapsed Time: 0:00:08 Time:  0:00:08\n","Buffers 100% (1 of 1) |#######################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Expected 100% (8434 of 8434) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-30 10:11:23,992 INFO Filtering low-coverage bins using absolute cutoff 0.0, relative cutoff 10.0%\n","2023-12-30 10:11:24,065 INFO Final absolute cutoff threshold is 872.9\n","2023-12-30 10:11:24,065 INFO Selected a total of 6 (4.3%) regions to be masked\n","2023-12-30 10:11:24,066 INFO Running filters...\n","Filter 100% (1 of 1) |########################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-30 10:11:24,108 INFO Total: 8434. Filtered: 143\n","Expected 100% (8291 of 8291) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-30 10:11:24,224 INFO Done.\n","2023-12-30 10:11:24,621 INFO Saving statistics...\n","2023-12-30 10:11:25,104 INFO Normalising binned Hic file\n","Expected 100% (8291 of 8291) |################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Closing remaining open files:test_necessaray_files_output/hic/fanc_example.hic...done\n","2023-12-30 10:11:27,384 INFO FAN-C version: 0.9.27\n","2023-12-30 10:11:27,496 INFO Binning Hic file (50000)\n","2023-12-30 10:11:27,496 INFO Constructing binned genome...\n","2023-12-30 10:11:28,048 INFO Binning edges...\n","2023-12-30 10:11:28,218 INFO Binning Hi-C contacts\n","2023-12-30 10:11:28,971 INFO Launching processes\n","2023-12-30 10:11:36,696 INFO Submitting partitions\n","2023-12-30 10:11:36,730 INFO Collecting results\n","Binning 100% (3 of 3) |#######################################| Elapsed Time: 0:00:05 Time:  0:00:05\n","Buffers 100% (6 of 6) |#######################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","Expected 100% (339969 of 339969) |############################| Elapsed Time: 0:00:02 Time:  0:00:02\n","2023-12-30 10:11:56,849 INFO Filtering low-coverage bins using absolute cutoff 0.0, relative cutoff 10.0%\n","2023-12-30 10:11:59,484 INFO Final absolute cutoff threshold is 44.45\n","2023-12-30 10:11:59,485 INFO Selected a total of 170 (6.2%) regions to be masked\n","2023-12-30 10:11:59,485 INFO Running filters...\n","Filter 100% (6 of 6) |########################################| Elapsed Time: 0:00:00 Time:  0:00:00\n","2023-12-30 10:12:00,618 INFO Total: 339969. Filtered: 1014\n","Expected 100% (338955 of 338955) |############################| Elapsed Time: 0:00:04 Time:  0:00:04\n","2023-12-30 10:12:05,272 INFO Done.\n","2023-12-30 10:12:06,039 INFO Saving statistics...\n","2023-12-30 10:12:06,697 INFO Normalising binned Hic file\n","Expected 100% (338955 of 338955) |############################| Elapsed Time: 0:00:02 Time:  0:00:02\n","Closing remaining open files:test_necessaray_files_output/hic/fanc_example.hic...done\n"]}]},{"cell_type":"markdown","source":["# test own data"],"metadata":{"id":"9HYl6sWwfCTP"}},{"cell_type":"code","source":["os.chdir(wd + \"/own_data\")\n","!pwd"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"upPRjTkeALss","executionInfo":{"status":"ok","timestamp":1704201886492,"user_tz":-480,"elapsed":7,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"5311e854-2bd0-412c-cf28-48217c6b9092"},"execution_count":13,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/NCCU_courses/112-1/Bioinformatics/readings/TADs/own_data\n"]}]},{"cell_type":"code","source":["# testing, output only processes\n","!fanc auto SRR5579160.fastq.gzip SRR5579160_1.fastq.gzip \\\n","    test_own_data_output/ \\\n","    -g dm6.fa \\\n","    -i dm6/dm6 \\\n","    -n own \\\n","    -t 4 \\\n","    --fanc-parallel \\\n","    -r HindIII \\\n","    --split-ligation-junction \\\n","    -q 30 \\\n","    -b 1mb, 50kb \\\n","    -f \\\n","    --run-with test"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"MOQi_YvefL3v","executionInfo":{"status":"ok","timestamp":1703953735147,"user_tz":-480,"elapsed":3020,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"c1f38c8e-5e69-4a60-be18-43c7cd6091c7"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["2023-12-30 16:28:53,470 INFO FAN-C version: 0.9.27\n","2023-12-30 16:28:54,524 INFO Output folder: test_own_data_output/\n","2023-12-30 16:28:54,524 INFO Input files: SRR5579160.fastq.gzip, SRR5579160_1.fastq.gzip\n","2023-12-30 16:28:54,524 INFO Input file types: fastq, fastq\n","2023-12-30 16:28:54,524 INFO Final basename: own (you can change this with the -n option!)\n","0 (threads: 4, depends on: ): fanc map -m 25 -s 3 -t 4 --fanc-parallel --no-iterative --restriction-enzyme HindIII SRR5579160.fastq.gzip dm6/dm6 test_own_data_output/sam/SRR5579160.bam\n","1 (threads: 4, depends on: ): fanc map -m 25 -s 3 -t 4 --fanc-parallel --no-iterative --restriction-enzyme HindIII SRR5579160_1.fastq.gzip dm6/dm6 test_own_data_output/sam/SRR5579160_1.bam\n","2 (threads: 1, depends on: 0, 1): fanc sort_sam -t 4 --no-sambamba test_own_data_output/sam/SRR5579160.bam\n","3 (threads: 1, depends on: 0, 1): fanc sort_sam -t 4 --no-sambamba test_own_data_output/sam/SRR5579160_1.bam\n","4 (threads: 4, depends on: 2, 3): fanc pairs -f -g dm6.fa -t 4 -us -r HindIII -q 30.0 -S test_own_data_output/sam/SRR5579160.bam test_own_data_output/sam/SRR5579160_1.bam test_own_data_output/pairs/own.pairs\n","5 (threads: 1, depends on: 4): fanc pairs -d 10000 -l -p 2 --statistics-plot test_own_data_output/plots/stats/own.pairs.stats.pdf test_own_data_output/pairs/own.pairs\n","6 (threads: 1, depends on: 5): fanc pairs --ligation-error-plot test_own_data_output/plots/stats/own.pairs.ligation_error.pdf test_own_data_output/pairs/own.pairs\n","7 (threads: 1, depends on: 5): fanc pairs --re-dist-plot test_own_data_output/plots/stats/own.pairs.re_dist.pdf test_own_data_output/pairs/own.pairs\n","8 (threads: 1, depends on: 5): fanc hic -f test_own_data_output/pairs/own.pairs test_own_data_output/hic/own.hic\n","9 (threads: 4, depends on: 8): fanc hic -f -b 1000000 -r 0.1 -t 4 --statistics-plot test_own_data_output/plots/stats/own_1mb.stats.pdf -n --norm-method kr test_own_data_output/hic/own.hic test_own_data_output/hic/binned/own_1mb.hic\n","10 (threads: 4, depends on: 8): fanc hic -f -b 50000 -r 0.1 -t 4 --statistics-plot test_own_data_output/plots/stats/own_50kb.stats.pdf -n --norm-method kr test_own_data_output/hic/own.hic test_own_data_output/hic/binned/own_50kb.hic\n"]}]},{"cell_type":"code","source":["!fanc auto SRR5579160.fastq.gzip SRR5579160_1.fastq.gzip \\\n","    test_own_data_output/ \\\n","    -g dm6.fa \\\n","    -i dm6/dm6 \\\n","    -n own \\\n","    -t 4 \\\n","    --fanc-parallel \\\n","    -r EcoRI \\\n","    --split-ligation-junction \\\n","    -q 30 \\\n","    -b 1mb, 50kb \\\n","    -f"],"metadata":{"id":"cUpXOzJwjo-e"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["```\n","2023-12-30 16:30:44,142 INFO FAN-C version: 0.9.27\n","2023-12-30 16:30:44,152 INFO Output folder: test_own_data_output/\n","2023-12-30 16:30:44,152 INFO Input files: SRR5579160.fastq.gzip, SRR5579160_1.fastq.gzip\n","2023-12-30 16:30:44,152 INFO Input file types: fastq, fastq\n","2023-12-30 16:30:44,152 INFO Final basename: own (you can change this with the -n option!)\n","2023-12-30 16:30:44,152 INFO Creating output folders...\n","2023-12-30 16:30:46,427 INFO FAN-C version: 0.9.27\n","2023-12-30 16:30:46,430 INFO Starting mapping for SRR5579160.fastq.gzip\n","2023-12-30 16:30:46,598 INFO Starting to output alignments to intermediate SAM file /tmp/tmpyw_euvu2/intermediate.sam\n","2023-12-30 17:01:50,036 INFO Converting intermediate SAM file to BAM (test_own_data_output/sam/SRR5579160.bam)\n","2023-12-30 17:02:54,345 INFO Success, removing intermediate.\n","2023-12-30 17:02:57,924 INFO FAN-C version: 0.9.27\n","2023-12-30 17:02:57,927 INFO Starting mapping for SRR5579160_1.fastq.gzip\n","2023-12-30 17:02:58,051 INFO Starting to output alignments to intermediate SAM file /tmp/tmp57lnsb2w/intermediate.sam\n","2023-12-30 20:22:48,861 INFO Converting intermediate SAM file to BAM (test_own_data_output/sam/SRR5579160_1.bam)\n","[E::bgzf_flush] File write failed (wrong size)\n","[E::bgzf_close] File write failed\n","2023-12-30 20:25:51,524 ERROR Could not convert to BAM, but your output is still in /tmp/tmp57lnsb2w/intermediate.sam\n","2023-12-30 20:25:58,824 INFO FAN-C version: 0.9.27\n","2023-12-30 20:25:58,825 INFO FAN-C version: 0.9.27\n","[E::hts_open_format] Failed to open file \"test_own_data_output/sam/SRR5579160_1.bam\" : Transport endpoint is not connected\n","Traceback (most recent call last):\n","  File \"/usr/local/bin/fanc\", line 127, in <module>\n","    Fanc()\n","  File \"/usr/local/bin/fanc\", line 93, in __init__\n","    command([sys.argv[0]] + sys.argv[option_ix:], log_level=log_level, verbosity=verbosity)\n","  File \"/usr/local/lib/python3.10/dist-packages/fanc/commands/fanc_commands.py\", line 647, in sort_sam\n","    output_file = sort_natural_sam(sam_file, output_file, threads=threads, sambamba=use_sambamba)\n","  File \"/usr/local/lib/python3.10/dist-packages/fanc/tools/files.py\", line 576, in sort_natural_sam\n","    pysam.sort('-n', '-o', output_file, sam_file)\n","  File \"/usr/local/lib/python3.10/dist-packages/pysam/utils.py\", line 83, in __call__\n","    raise SamtoolsError(\n","pysam.utils.SamtoolsError: 'samtools returned with error 1: stdout=, stderr=samtools sort: can\\'t open \"test_own_data_output/sam/SRR5579160_1.bam\": Transport endpoint is not connected\\n'\n","[E::hts_open_format] Failed to open file \"test_own_data_output/sam/SRR5579160.bam\" : Transport endpoint is not connected\n","Traceback (most recent call last):\n","  File \"/usr/local/bin/fanc\", line 127, in <module>\n","    Fanc()\n","  File \"/usr/local/bin/fanc\", line 93, in __init__\n","    command([sys.argv[0]] + sys.argv[option_ix:], log_level=log_level, verbosity=verbosity)\n","  File \"/usr/local/lib/python3.10/dist-packages/fanc/commands/fanc_commands.py\", line 647, in sort_sam\n","    output_file = sort_natural_sam(sam_file, output_file, threads=threads, sambamba=use_sambamba)\n","  File \"/usr/local/lib/python3.10/dist-packages/fanc/tools/files.py\", line 576, in sort_natural_sam\n","    pysam.sort('-n', '-o', output_file, sam_file)\n","  File \"/usr/local/lib/python3.10/dist-packages/pysam/utils.py\", line 83, in __call__\n","    raise SamtoolsError(\n","pysam.utils.SamtoolsError: 'samtools returned with error 1: stdout=, stderr=samtools sort: can\\'t open \"test_own_data_output/sam/SRR5579160.bam\": Transport endpoint is not connected\\n'\n","Traceback (most recent call last):\n","  File \"/usr/local/bin/fanc\", line 127, in <module>\n","    Fanc()\n","  File \"/usr/local/bin/fanc\", line 93, in __init__\n","    command([sys.argv[0]] + sys.argv[option_ix:], log_level=log_level, verbosity=verbosity)\n","  File \"/usr/local/lib/python3.10/dist-packages/fanc/commands/fanc_commands.py\", line 123, in auto\n","    return fanc.commands.auto.auto(argv, **kwargs)\n","  File \"/usr/local/lib/python3.10/dist-packages/fanc/commands/auto.py\", line 1311, in auto\n","    runner.run()\n","  File \"/usr/local/lib/python3.10/dist-packages/fanc/commands/auto.py\", line 141, in run\n","    raise RuntimeError(\"Task {} had non-zero exit status. \"\n","RuntimeError: Task e47ea327-1f25-4e45-bace-c8225d5d42e4 had non-zero exit status. Cancelling execution of all tasks.\n","```"],"metadata":{"id":"nrWeII3uRWw3"}},{"cell_type":"markdown","source":["# normalise"],"metadata":{"id":"-vivMoJKVfUO"}},{"cell_type":"code","source":["import fanc\n","import fanc.plotting as fancplot\n","\n","# Load your Hi-C data\n","hic = fanc.load(\"test_own_data_output/hic/SRR5579163_A_hic.hic\")\n","\n","# Normalize the Hi-C data\n","hic_norm = hic.normalise(method=\"KR\")\n","\n","# Save the normalized Hi-C data\n","hic_norm.save(\"test_own_data_output/hic/SRR5579163_A_norm.hic\")"],"metadata":{"id":"D889DdXQVDTl"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# !fancplot chr2L:0mb-43347mb -o theplot_hy.png -p square -vmax 0.1 test_own_data_output/hic/SRR5579163_A_hic.hic\n","!fancplot chr2L:0mb-43347mb -o theplot_hy.png -p square -vmax 0.05 test_own_data_output/hic/SRR5579163_A_hic.hic"],"metadata":{"id":"EfFRY-RskD8l","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1704201558192,"user_tz":-480,"elapsed":15895,"user":{"displayName":"何子安","userId":"06861626017998849506"}},"outputId":"bc08ba82-e787-4ba6-fd38-7a3b4618ec74"},"execution_count":15,"outputs":[{"output_type":"stream","name":"stdout","text":["2024-01-02 13:19:04,338 INFO Using non-interactive backend\n","2024-01-02 13:19:05,248 INFO Found 1 regions\n","Closing remaining open files:test_own_data_output/hic/SRR5579163_A_hic.hic...done\n"]}]},{"cell_type":"code","source":[],"metadata":{"id":"T1Nno4AWUeRy"},"execution_count":null,"outputs":[]}]}